app/src/main/java/org/mozilla/fenix/home/collections/CollectionViewHolder.kt

```diff
@@ -0,0 +1,194 @@
+        this.collection = collection
+        isExpanded = expanded
+    }
+
+    companion object {
+        val LAYOUT_ID = View.generateViewId()
+        const val maxTitleLength = 20
+    }
+}
+
+/**
+ * Rectangular shape with all corners rounded used to display a collapsed collection.
+ */
+private val collapsedCollectionShape = RoundedCornerShape(8.dp)
+
+/**
+ * Rectangular shape with only the top corners rounded used to display an expanded collection with other views
+ * placed immediately below this which can be shown immediately next to it, with no visible separation.
+ */
+private val expandedCollectionShape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp)
+
+@Composable
+private fun Collection(
+    collection: TabCollection,
+    expanded: Boolean,
+    menuItems: List<CollectionMenuItemData>,
+    interactor: CollectionInteractor,
+) {
+    var isMenuShown by remember { mutableStateOf(false) }
+    val isExpanded by remember { mutableStateOf(expanded) }
+
+    Card(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .clickable { interactor.onToggleCollectionExpanded(collection, !isExpanded) }
+            .height(48.dp),
+        shape = if (isExpanded) expandedCollectionShape else collapsedCollectionShape,
+        backgroundColor = FirefoxTheme.colors.layer2,
+        elevation = 5.dp, // This needs to match the elevation of the TabInCollectionViewHolder for the shadow.
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxHeight(),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                painter = painterResource(R.drawable.ic_tab_collection),
+                contentDescription = null,
+                modifier = Modifier.padding(
+                    start = 16.dp,
+                    end = 8.dp // (24.dp - 16.dp) hardcoded in ExpandableListHeader
+                ),
+                tint = Paint().apply {
+                    color = Color(collection.getIconColor(LocalContext.current))
+                    blendMode = BlendMode.SrcIn
+                }.color
+            )
+
+            ExpandableListHeader(
+                headerText = collection.title,
+                expanded = isExpanded,
+            ) {
+                if (isExpanded) {
+                    Row {
+                        IconButton(
+                            onClick = {
+                                interactor.onCollectionShareTabsClicked(
+                                    collection
+                                )
+                            }
+                        ) {
+                            Icon(
+                                painter = painterResource(R.drawable.ic_share),
+                                contentDescription = stringResource(R.string.share_button_content_description),
+                                tint = FirefoxTheme.colors.textPrimary
```

Comment:
I believe this should be `FirefoxTheme.colors.iconPrimary`

Reply:
Thank you!
`textPrimary` was [the color used in xml](https://github.com/mozilla-mobile/fenix/blob/f924f744348006e2cb01fb172486fc7bfb1b67dc/app/src/main/res/drawable/ic_share.xml#L11).
`FirefoxTheme.colors.iconPrimary` seems to have the same values as `FirefoxTheme.colors.textPrimary` and does fit better in the context, will change to that.